okta/resource_okta_app_oauth.go

```diff
@@ -770,21 +770,31 @@ func buildAppOAuth(d *schema.ResourceData) *sdk.OpenIdConnectApplication {
 	}
 
 	refresh := &sdk.OpenIdConnectApplicationSettingsRefreshToken{}
-	hasRefresh := false
+	var hasRefresh bool
+	for _, grant_type := range grantTypes {
+		if grant_type == refreshToken {
+			hasRefresh = true
+		}
+	}
 	if rotate, ok := d.GetOk("refresh_token_rotation"); ok {
 		refresh.RotationType = rotate.(string)
-		hasRefresh = true
 	}
 
 	if leeway, ok := d.GetOk("refresh_token_leeway"); ok {
 		refresh.LeewayPtr = int64Ptr(leeway.(int))
-		hasRefresh = true
 	}
 
 	if hasRefresh {
 		app.Settings.OauthClient.RefreshToken = refresh
 	}
 
+	// TODU
+	// if !hasRefresh && refresh != nil {
+	// 	return nil, errors.New("does not have refresh grant type but refresh_token_rotation and refresh_token_leeway exist in payload")
+	// }
+
+	// TODU
+	// should we unset here refresh_token_rotation, refresh_token_leeway?
```

Comment:
Should we unset refresh_token_rotation and refresh_token_leeway from the resource if it does not include the refresh?